pkg/pod/pod.go

```diff
@@ -224,6 +226,10 @@ func MakePod(images pipeline.Images, taskRun *v1alpha1.TaskRun, taskSpec v1alpha
 	podAnnotations := taskRun.Annotations
 	podAnnotations[ReleaseAnnotation] = ReleaseAnnotationValue
 
+	if shouldAddReadyAnnotationonPodCreate(taskSpec.Sidecars, kubeclient) {
+		podAnnotations[readyAnnotation] = readyAnnotationValue
```

Comment:
it feels a bit odd to me that we are using `readyAnnotation` directly here AND we're using it in `UpdateReady` in a totally different file - could it make sense to call `UpdateReady` here, or in some way refactor this a bit so we can share the code that accesses `readyAnnotation`?

Reply:
It is a different file but in the same package though. I don't think we can call `UpdateReady` here because that is updating an already existing Pod while the usage here is adding to a Pod's definition before it is created.
One thing we could do is move this and UpdateReady to its own file in the same package...but that puts UpdateReady in a different file from all the other Sidecar functions in that file.